sat/cli/bootprep/input/configuration.py

```diff
@@ -25,22 +25,48 @@
 Defines classes for configurations defined in the input file.
 """
 from abc import ABC, abstractmethod
+import functools
 import logging
 from urllib.parse import urlparse, urlunparse
 
 from cray_product_catalog.query import ProductCatalogError
+from jinja2 import Template, TemplateError
 
 from sat.apiclient.vcs import VCSError, VCSRepo
 from sat.cached_property import cached_property
+from sat.cli.bootprep.constants import LATEST_VERSION_VALUE
 from sat.cli.bootprep.errors import ConfigurationCreateError
 from sat.config import get_config_value
 
-# This value is used to specify that the latest version of a product is desired
-LATEST_VERSION_VALUE = 'latest'
-
 LOGGER = logging.getLogger(__name__)
 
 
+def render_template(func):
+    """Wrapper function to handle Jinja templates with variables stored in object.
+
+    This should be used to wrap instance methods which return content from the
+    InputInstance that may contain Jinja2 template content to be rendered.
+
+    This wrapper assumes the instance has a `var_context` attribute of type
+    `VariableContext` that should be used for rendering templates.
+    """
+    @functools.wraps(func)
+    def wrapper(self, *args, **kwargs):
+        unrendered_result = func(self, *args, **kwargs)
+
+        # If the value is `None`, it was not specified by the user.
+        if unrendered_result is None:
+            return unrendered_result
+
+        try:
+            return Template(unrendered_result).render(self.var_context.vars)
```

Comment:
I don't remember if you mentioned this in the testing (Jira is down at the moment), but if a variable is missing from `self.var_context.vars`, will this just replace it with the empty string? e.g. if the input template has the following somewhere:
```version: {{misspelled_product_name.version}}```
Would that just result in the following, which is semantically invalid?
```layer.product_version == ""```
If this is the case, would it make sense to pass `undefined=jinja2.StrictUndefined` to the `Template()` constructor, and handle resulting [`UndefinedError`](https://jinja.palletsprojects.com/en/3.1.x/api/#jinja2.StrictUndefined)s?

Reply:
Summarizing the discussion from scrum, if we try to access a key in an undefined variable, e.g. `version: {{misspelled_product_name.version}}`, we will get an error. If we access just a variable that doesn't exist, it will be replaced by an empty string. I think that's probably okay. The dotted variable paths are the more common use case right now.

Reply:
I will be giving this more thought and testing as I go on an implement other variables though. I added a note to CRAYSAT-1411 to make sure I consider this then when I'm implementing additional non-dict variables.